templates/junit4.java.qute

```diff
@@ -0,0 +1,54 @@
+///usr/bin/env jbang "$0" "$@" ; exit $?
+
+//DEPS junit:junit-dep:4.11
+
+//SOURCES {scriptName}.java
+
+import org.junit.Test;
+import org.junit.runner.JUnitCore;
+import org.junit.runner.Request;
+import org.junit.runner.Result;
+import org.junit.runner.notification.Failure;
+
+import static java.lang.System.out;
+
+// JUnit4 Test class for {scriptName}
+public class {baseName} {
```

Comment:
Should this not be `{baseName}Test` ?

Reply:
Depends how you do the init.

My assumption is to pass to the `init` command the test class as an argument and the class to test as an init parameter like this:

`jbang init -DscriptName=Script --template=junit5@jbangdev ScriptTest.java `

@quintesse proposed to simplify it by passing the class to test only and append Test to create the test class but i don't get it to work :-(

To say it shortly, the created file is always {basename} so if i do:

`jbang init --template=junit5@jbangdev Script`

`jbang` creates a Script.java file which is not what i expect !

Please see previous Thread and tell me how i can modify the .qute file and templates to make it work.

Reply:
Ack. You'll need latest 0.79 to Do the script ref trick @quintesse refers to. 

I'll try update when I'm back at laptop.

Reply:
It works using:

`
      "file-refs": {
        "{basename}Test.java": "templates/junit4b.java.qute"
      },
`

with:

`jbang init --verbose --fresh --template=junit4b@jbangdev_local Script`

and using only `{scriptref}` in template but `jbang` output is puzzling:

`
[jbang] Rendering template /home/pyfourmond/Documents/Sources/mine/jbangdev-jbang-catalog/templates/junit4b.java.qute to /home/pyfourmond/Documents/Sources/mine/jbang-scripts/**ScriptTest.java**
[jbang] File initialized. You can now run it with 'jbang Script' or edit it using 'jbang edit --open=[editor] **Script**' where [editor] is your editor or IDE, e.g. 'vi'
`

The suggested edit command is incorrect.

Reply:
> The suggested edit command is incorrect.

That's indeed something that would have to be fixed.  Right now it's assumed that the name you give is the name that's going to be generated, In this case we'd have to be a little smarter about it.

Checking in with you, @maxandersen , to confirm that this is really what we want to do. I know it was my idea in the first place but it might be somewhat surprising that the name you type is not what gets generated (especially  in the case of `jbang init --template=junit5@jbangdev Script.java`). So I'm wondering if the surprise is worth the simplification of the command line invocation. Wdyt?

Reply:
I think it's fine and we even support both approaches. I'll try update the pr. 

For the edit command printed we need to fix that anyway as it assumes template doesn't rename the file. So that's a bug that should get fixed to take the first output name.

Reply:
I've opened issue #1019 for the incorrect output.